src/System.Net.Security/src/System/Net/Security/SNIHelper.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Net.Security
+{
+    internal class SNIHelper
```

Comment:
nit: SNIHelper -> SniHelper
Style guideline uses normal Pascal casing when acronym is more than 2 characters.  Change filename as well to match.